### PR TITLE
Feature/trolley api

### PR DESCRIPTION
**Pull request by AussieKing**
JIRA TICKET # to be added here 

tagging you to check branch  feature/trolley-api work.
API calls now returning right status and messages.
Acceptance Criteria for Story 1 now met.
@ssu8

### CODE REVIEW

File: Controllers/BeersControllers.cs [hunk 1]
```diff
@@ -85,8 +75,6 @@ public ActionResult UpdatePromoPrice(int id, [FromBody] PromoPriceUpdateRequest
             {
                 return BadRequest("Promotional price cannot be less than half of the normal price."); 
             }
-
-            beer.PromoPrice = request.NewPromoPrice;
```
Comment (ssu8) [inline Controllers/BeersControllers.cs:89]: Have you broken functionality here? This method is no longer updating the promo price. It validates the promo price and that's it.

File: Controllers/TrolleyController.cs [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        private readonly ITrolleyService _trolleyService;
+        private readonly IBeerService _beerService;
+
+        public TrolleyController(ITrolleyService trolleyService, IBeerService beerService)
+        {
+            _trolleyService = trolleyService;
+            _beerService = beerService;
+        }
+
+        [HttpPost("add/{beerId}")]
```
Comment (ssu8) [inline Controllers/TrolleyController.cs:21]: Remember this should be a RESTful API. Best practice for REST is to use nouns in the route, and avoid verbs. Have a think about how this route can be expressed more RESTfully. 
* [Here](https://swagger.io/resources/articles/best-practices-in-api-design/) is a short article on API design and practices
* The `BeerController` is RESTful, see if you can draw inspiration from that.

File: Controllers/TrolleyController.cs [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+        public IActionResult AddItemToTrolley(int beerId)
+        {
+            var beer = _beerService.GetBeerById(beerId);
+            if (beer == null)
+            {
+                return NotFound($"Beer with ID {beerId} not found.");
+            }
+
+            _trolleyService.AddItem(beer);
```
Comment (ssu8) [inline Controllers/TrolleyController.cs:30]: This is alright if my website only has 1 customer. What happens if there's many customers trying to add items to their trolley?

File: Controllers/TrolleyController.cs [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+            return Ok(new { message = "Item added to trolley.", trolley = _trolleyService.GetTrolley() });
+        }
+
+        [HttpPost("remove/{beerId}")]
```
Comment (ssu8) [inline Controllers/TrolleyController.cs:34]: This one as well. Have a think about making this more RESTful.

File: bin/Debug/net8.0/BeerAPI.dll
Comment (ssu8) [inline bin/Debug/net8.0/BeerAPI.dll:1]: See if you can clean up this repo by removing the binary files.

File: Controllers/TrolleyController.cs (hunk 2 above)
Comment (AussieKing) [reply Controllers/TrolleyController.cs:21]: Of course.
[HttpPost("{beerId}")] // To add an item
[HttpDelete("{beerId}")] // To remove an item

File: Controllers/TrolleyController.cs (hunk 4 above)
Comment (AussieKing) [reply Controllers/TrolleyController.cs:34]: [HttpDelete("{beerId}")] // To remove an item

File: Program.cs [hunk 5]
```diff
@@ -13,7 +13,10 @@
 builder.Services.AddControllers(); 
 builder.Services.AddFluentValidationAutoValidation(); 
 builder.Services.AddTransient<IValidator<Beer>, BeerValidator>(); 
-builder.Services.AddScoped<IBeerDescriptionService, BeerDescriptionService>(); 
+builder.Services.AddScoped<IBeerDescriptionService, BeerDescriptionService>();
+builder.Services.AddScoped<IBeerService, BeerService>();
+builder.Services.AddScoped<ITrolleyService, TrolleyService>();
+builder.Services.AddSingleton<ITrolleyService, TrolleyService>();
```
[the remark this answers is not in the document]
Comment (AussieKing) [reply Program.cs:19]: deleted the Singleton

**Pull request merged**